modules/tinymce/src/core/main/ts/newline/InsertBlock.ts

```diff
@@ -20,71 +21,66 @@ import * as NewLineUtils from './NewLineUtils';
 const trimZwsp = (fragment: DocumentFragment) => {
   Arr.each(PredicateFilter.descendants(SugarElement.fromDom(fragment), SugarNode.isText), (text) => {
     const rawNode = text.dom;
-    rawNode.nodeValue = Zwsp.trim(rawNode.nodeValue);
+    rawNode.nodeValue = Zwsp.trim(rawNode.data);
   });
 };
 
-const isEmptyAnchor = (dom: DOMUtils, elm: Element) => {
+const isEmptyAnchor = (dom: DOMUtils, elm: Node): boolean => {
   return elm && elm.nodeName === 'A' && dom.isEmpty(elm);
 };
 
-const isTableCell = (node) => {
```

Comment:
Made this use the `NodeType.isTableCellOrCaption` function instead to remove some duplication.